pkg/apis/autoscaling/v1alpha1/hpaaggregator_types.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2023 The Karmada Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	autoscalingv2 "k8s.io/api/autoscaling/v2"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// HPAAggregator is a centralized HPA aggregator that can aggregate the metrics in multiple clusters.
+// When the system load increases, it will query the metrics from multiple clusters and scales up the replicas.
+// When the system load decreases, it will query the metrics from multiple clusters and scales down the replicas.
+// After the replicas are scaled up/down, karmada-scheduler will schedule the replicas based on the policy.
+type HPAAggregator struct {
+	metav1.TypeMeta
+	metav1.ObjectMeta
+
+	// Spec is the specification of the HPAAggregator.
+	// +required
+	Spec HPAAggregatorSpec `json:"spec"`
```

Comment:
Here we do not use the original HPA spec is for extending new fields in the future, right?

Reply:
Yes,